engine.go

```diff
@@ -334,6 +373,51 @@ func (e *Engine) analyzeQuery(ctx *sql.Context, query string, parsed sql.Node, b
 		return nil, err
 	}
 
+	switch n := parsed.(type) {
+	case *plan.PrepareQuery:
+		analyzedChild, err := e.Analyzer.PrepareQuery(ctx, n.Child, nil)
+		if err != nil {
+			return nil, err
+		}
+		e.PreparedDataCache.CacheStmt(ctx.Session.ID(), n.Name, analyzedChild)
+	case *plan.ExecuteQuery:
+		// replace execute query node with the one prepared
+		p, ok := e.PreparedDataCache.GetCachedStmt(ctx.Session.ID(), n.Name)
+		if !ok {
+			return nil, sql.ErrUnknownPreparedStatement.New(n.Name)
+		}
+
+		// number of BindVars provided must match number of BindVars expected
+		if countBindVars(p) != len(n.BindVars) {
+			return nil, sql.ErrInvalidArgument.New(n.Name)
+		}
+		parsed = p
+
+		bindings = map[string]sql.Expression{}
+		for i, binding := range n.BindVars {
+			varName := fmt.Sprintf("v%d", i+1)
+			bindings[varName] = binding
+		}
+
+		if len(bindings) > 0 {
+			parsed, err = plan.ApplyBindings(parsed, bindings)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		analyzed, _, err = e.Analyzer.AnalyzePrepared(ctx, parsed, nil)
+		if err != nil {
+			return nil, err
+		}
+		return analyzed, nil
+	case *plan.DeallocateQuery:
+		if _, ok := e.PreparedDataCache.GetCachedStmt(ctx.Session.ID(), n.Name); !ok {
+			return nil, sql.ErrUnknownPreparedStatement.New(n.Name)
+		}
+		e.PreparedDataCache.UncacheStmt(ctx.Session.ID(), n.Name)
```

Comment:
should this return on completion?

Reply:
Should the function `UncacheStmt` return something? or should the switch statement return something?
I could return in each case of the switch, since the analyzer has no rules for `PrepareQuery`, `ExecuteQuery`, or `DeallocateQuery`, so there's no reason to analyze them.

Reply:
Yup; I read that as "returning in each case of the switch" (like you're already doing for ExecuteQuery)